src/sentry/remote_config/docs/protocol.md

```diff
@@ -22,17 +22,68 @@ Retrieve a project's configuration.
+| operator | Operator                                       | A conditional operator that evaluates the value against the target.     |
+| property | string                                         | A context object property name which contains the value of the operand. |
+| value    | union[string, number, bool, null, array[self]] | The value being compared against.                                       |
+
+Rules contain evaluation and extraction instructions. The context object provided must contain the key defined on the `property` attribute. If the SDK can not find the `property` within the context object the evaluation fails and the variant is skipped. The `operator` defines how the context value is compared against the literal provided by the configuration on the `value` attribute. If the value provided in the context object can not be compared against the value provided in the configuration the evaluation fails and the variant is skipped.
+
+The type of the `value` field is controlled by the operator being applied to it. For example, some operators only operate on `number` types. While others operate on arrays.
+
+**Operator Reference**
+
+| Symbol | Value Type                         | Description                                                               |
+| ------ | ---------------------------------- | ------------------------------------------------------------------------- |
+| ==     | union[string, bool, number]        | Assert the two values match.                                              |
+| !=     | union[string, bool, number]        | Assert the two values do not match.                                       |
+| >=     | number                             | Assert the local value is greater than or equal to the remote value.      |
+| >      | number                             | Assert the local value is greater than the remote value.                  |
+| <=     | number                             | Assert the local value is less than or equal to the remote value.         |
+| <      | number                             | Assert the local value is less than the remote value.                     |
+| IN     | array[union[string, bool, number]] | Assert the local value is contained within an array of remote values.     |
+| NOT IN | array[union[string, bool, number]] | Assert the local value is not contained within an array of remote values. |
 
 **Option Object**
 
-| Field              | Type  | Description        |
-| ------------------ | ----- | ------------------ |
-| sample_rate        | float | Error sample rate. |
-| traces_sample_rate | float | Trace sample rate. |
+| Field              | Type                 | Description        |
+| ------------------ | -------------------- | ------------------ |
+| sample_rate        | OptionVariant<float> | Error sample rate. |
+| traces_sample_rate | OptionVariant<float> | Trace sample rate. |
```

Comment:
Is `OptionVariant` meant to be our `0.0 to 1.0` type. Could slap that into the top and re-use the type name for `rollout`.

Reply:
oh wait, `Variant` is the name of the existing type above... but in the example it looks like `sample_rate` is using that type.

therefore, we should use the same name here:

```suggestion
| sample_rate        | Variant<float> | Error sample rate. |
| traces_sample_rate | Variant<float> | Trace sample rate. |
```

Reply:
I need to document it but basically all of the sentry options should have the same feature set as the items in the `features` array.  Except since they're statically typed I'm trying to communicate that its generic over type <T> and the sample_rates are floats.  This part needs work.

Reply:
Except its missing the `key` field.  So its not exactly the same type.

Reply:
Added option feature docs.

Reply:
Turns out its super easy to share code since you're just dropping the key and reusing the generic variants structure.

Reply:
sweet